### ISSUE TITLE
[BUG] : Drawer Not Functioning Properly in App Scaffold

### ISSUE THREAD

**Issue by may-tas** [Jiggy9/JCT#66]
## BUG : 

- The app's scaffold drawer is not opening as expected when interacting with the drawer icon. 
- Despite the drawer property being correctly assigned, the drawer fails to appear or respond to user interactions. 
- This issue affects navigation and user accessibility within the app.

**Comment by may-tas** [comment]
Please assign me this issue.

**Comment by Jiggy9** [comment]
> Please assign me this issue.

Go ahead

**Comment by CodewithAnn** [comment]
> ## BUG :
> * The app's scaffold drawer is not opening as expected when interacting with the drawer icon.
> * Despite the drawer property being correctly assigned, the drawer fails to appear or respond to user interactions.
> * This issue affects navigation and user accessibility within the app.

there is nothing to do with it, there was a drawer but it was removed.

**Comment by may-tas** [comment]
> > ## BUG :
> > 
> > * The app's scaffold drawer is not opening as expected when interacting with the drawer icon.
> > * Despite the drawer property being correctly assigned, the drawer fails to appear or respond to user interactions.
> > * This issue affects navigation and user accessibility within the app.
> 
> there is nothing to do with it, there was a drawer but it was removed.

No, there is still a widget main_drawer.dart and it's implemented in the Important_screen.dart.
It's just the implementation for opening the drawer through icon button is wrong , which I will correct.

**Comment by Jiggy9** [comment]
> > > ## BUG :
> > > 
> > > * The app's scaffold drawer is not opening as expected when interacting with the drawer icon.
> > > * Despite the drawer property being correctly assigned, the drawer fails to appear or respond to user interactions.
> > > * This issue affects navigation and user accessibility within the app.
> > 
> > 
> > there is nothing to do with it, there was a drawer but it was removed.
> 
> No, there is still a widget main_drawer.dart and it's implemented in the Important_screen.dart. It's just the implementation for opening the drawer through icon button is wrong , which I will correct.

Could you send ss of it

**Comment by may-tas** [comment]
![Screenshot_2024-10-14-16-05-54-92_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/ba6ee31e-7131-4876-885a-d5645265c2df)
![Screenshot_2024-10-14-16-06-13-74_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/342b326b-d3f6-4082-9899-58fb8b97b115)

**Comment by CodewithAnn** [comment]
@may-tas  bro what  did you in last pr #65

**Comment by CodewithAnn** [comment]
> > > ## BUG :
> > > 
> > > * The app's scaffold drawer is not opening as expected when interacting with the drawer icon.
> > > * Despite the drawer property being correctly assigned, the drawer fails to appear or respond to user interactions.
> > > * This issue affects navigation and user accessibility within the app.
> > 
> > 
> > there is nothing to do with it, there was a drawer but it was removed.
> 
> No, there is still a widget main_drawer.dart and it's implemented in the Important_screen.dart. It's just the implementation for opening the drawer through icon button is wrong , which I will correct.

just remove on implementation no need of drawer anymore there is bottom nav now

**Comment by may-tas** [comment]
There was icon button and code still made for drawer and it was not working, hence I thought it's necessary, if it isn't @Jiggy9 you can close this issue.

**Comment by CodewithAnn** [comment]
> There was icon button and code still made for drawer and it was not working, hence I thought it's necessary, if it isn't @Jiggy9 you can close this issue.

let me ask you 2question?

1. Why you delete provider and use cubit 
2. why use remove bottom navigation and use package to handle it (18 months old) buddy not good practice

**Comment by may-tas** [comment]
1. Cubit is better and the implementation and code is simpler you can read my PR for more details.
2. using a package is simpler that's why I used it, I know the package is old but that can be changed to another package , it's better than writing long code for UI

**Comment by CodewithAnn** [comment]
> 1. Cubit is better and the implementation and code is simpler you can read my PR for more details.
> 2. using a package is simpler that's why I used it, I know the package is old but that can be changed to another package , it's better than writing long code for UI

your code is going to revert if the bug you produced not solved by me.  
Having package isn't mean that it should be use case if we can solve a problem without using the package, code should be package independent " I am saying from experience" . And about cubit you did nothing in it just another stly of write same code in cubit it just update the index of the navv bar so the provider as well. An advice take it or leave it: Never be afraid to write code there are no shortcuts.

**Comment by may-tas** [comment]
I don't always use a package, I've contributed a fair share in other repositories and I know that writing code is not something to be afraid of, I just don't know why are you arguing with me on such a small topic.
I know cubit is looking very simple because it is, plus the main thing is I've merged the logic of bottom_navigation.dart into the important_screen.dart plus I also refactored the code so that It looks readable.
Btw I do agree that native code is sometimes better but I think packages also make the code simpler and beginner friendly.
That's all.
